src/Tasks.js

```diff
@@ -0,0 +1,496 @@
+let statusColorMapping = ["yellow" , "purple" , "red" , "chartreuse"];
+let mapStatus = {
+    "Not started" : 0,
+    "In-progress" : 1,
+    "On hold" : 2,
+    "Completed" : 3,
+}
+
+
+function getDateString(date) {
+    let dateString = " ";
+    dateString+=date.slice(8,10);
+    dateString+=" ";
+    dateString+=monthMapping[+date.slice(5,7)-1];
+    dateString+=",";
+    dateString+=date.slice(0,4);
+    return dateString;
+}
+
+function getStageString(stages) {
+    let totalStages = 0;
+    let completedStages = 0;
+    for(let prop in stages)
+    {
+        totalStages++;
+        if(stages[prop])
+        {
+            completedStages++;
+        }
+    }
+    let stagesString = " " + completedStages + "/" + totalStages;
+    return stagesString;
+}
+
+function TaskInfo({task}) {
+    return(
+        <div className="taskInfo">
+            <span className="taskStatus">
+                <i className="fa fa-square taskStatusSymbol" aria-hidden="true" style={{color : statusColorMapping[task.status]}}></i> {statusMapping[task.status]}
+            </span>
+            <span className="taskDue">
+                <i className="fa fa-clock-o" aria-hidden="true"></i> {getDateString(task.dueDate)}
+            </span>
+            <span>
+                <i className="fa fa-check-square-o" aria-hidden="true"></i> {getStageString(task.stages)}
+            </span>
+        </div>
+    );
+}
+
+const TaskImage = React.memo(function({src}) {
+    return (
+        <div className="taskImage">
+            <img src = {src} height="160px" width="252px"/>
+        </div>
+    );
+})
+
+const TaskCard = React.memo(function({task,editClick}) {
+
+    function editCallback() {
+        editClick(task);
+    }
+
+    return (
+        <div className="card" onClick={editCallback}>
+            <TaskImage src={task.imageUrl} />
+            <div className="taskTitle">
+                {task.title}
+            </div>
+            <hr className="titleLineBreaks" />
+            <TaskInfo task={task} />
+        </div>
+    )
+});
+
+function UserColumn({userTasks,userName,editClickHandle,userId}) {
+    console.log(userTasks);
+
+    const editClick = useCallback((task) => editClickHandle(task),[]);
+
+    return (
+        <section className="board-list">
+            <div className="tasksAdd" onClick={() => editClick({assignee: userId,imageUrl: "assets/default.png",status:0})}>
+                <i className="fa fa-plus-circle" aria-hidden="true"></i>
+            </div>
+            <div className="list-title">
+                {userName}
+            </div>
+            {userTasks.map(task => <TaskCard task={task} key={task.taskId} editClick={editClick}/>)}
+        </section>
+    );
+}
+
+function StagesDisplay({stages,closeStage}) {
+    let stageArray = [];
+    for(let stage in stages) {
+        stageArray.push(stage);
+    }
+
+    function handleChange(e) {
+        if(e.target.value == "on")
```

Comment:
`if(e.target.value == "on")
        {
            stages[e.target.dataset.stagename] = true;
        }
        else
        {
            stages[e.target.dataset.stagename] = false;
        }`

can we write this in a single expression without using if/else?